app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt

```diff
@@ -190,11 +190,11 @@ class EndToEndTest {
         .onNodeWithTag("DescriptionField")
         .performTextInput("Teach me how to write tests pls")
     composeTestRule.onNodeWithTag("DateButton").performClick()
-    Thread.sleep(2000)
-    onView(withText("OK")).perform(click())
+    composeTestRule.waitUntil(15000) { composeTestRule.onNodeWithText("OK").isDisplayed() }
+    composeTestRule.onNodeWithText("OK").performClick()
     composeTestRule.onNodeWithTag("TimeButton").performClick()
-    Thread.sleep(2000)
-    onView(withText("OK")).perform(click())
+    composeTestRule.waitUntil(15000) { composeTestRule.onNodeWithText("OK").isDisplayed() }
+    composeTestRule.onNodeWithText("OK").performClick()
```

Comment:
I see you copied the new date and time pickers from my branch, that should be helpful for CI reliability!